app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/AgendaScreen.kt

```diff
@@ -614,6 +625,39 @@ fun AgendaScreenContent(
         )
     }
 
+    // • Select current date for agenda
+    state.chooseCurrentDateDialog ?.let { currentDate ->
+
+        var pickedDate by remember(currentDate) { mutableStateOf(currentDate) }
+        val dateDialogState = rememberMaterialDialogState()
+
+        dateDialogState.show()
+        MaterialDialog(
+            dialogState = dateDialogState,
+            onCloseRequest = {
+                dateDialogState.hide()
+                onAction(CancelChooseCurrentDateDialog)
+            },
+            buttons = {
+                positiveButton(text = stringResource(R.string.ok)) {
+                    dateDialogState.hide()
+                    onAction(SetCurrentDate(pickedDate))
+                }
+                negativeButton(text = stringResource(R.string.cancel)) {
+                    dateDialogState.hide()
+                    onAction(CancelChooseCurrentDateDialog)
+                }
+            }
+        ) {
```

Comment:
This looks like a side effect. `dateDialogState.show()` is not a composable function, but executed in a composable context. You can safely put the dialog composable anywhere at the root of your `AgendaScreen` composable and show it like this:
```
LaunchedEffect(state.chooseCurrentDateDialog) {
    state.chooseCurrentDateDialog?.let { dateDialogState.show() }
}
```

Reply:
OK, I will refactor this.